Samples/NetworkMediaStreamer/NetworkMediaStreamerBase/inc/NwMediaStreamSinkBase.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for more information
+
+#pragma once
+
+#define RETURNIFSHUTDOWN if(m_bIsShutdown) return MF_E_SHUTDOWN;
+
+#define HRESULT_EXCEPTION_BOUNDARY_FUNC catch(...) { auto hr = winrt::to_hresult(); return hr;}
+
+class NwMediaStreamSinkBase : public winrt::implements<NwMediaStreamSinkBase, INetworkMediaStreamSink, IMFStreamSink, IMFMediaEventGenerator>
+{
+protected:
+    uint8_t* m_pVideoHeader;
+    uint32_t m_VideoHeaderSize;
+    bool m_bIsShutdown;
+    IMFMediaSink* m_pParentSink;

```

Comment:
you can use weakref, it can avoid the circular reference

Reply:
there is no circular reference as the streamsink doesn't AddRef to parent. We dont need the ref because the streamsinks will be allocated and deallocated by the parent and the parent lifetime is guaranteed for the streamsink.

---
In reply to: [494099805](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r494099805) [](ancestors = 494099805)

Reply:
i meant to reference the comment you have in the constructor which explains you are storing this as raw pointer to avoid circular reference.  just a nit: you can pass the parent in as weak_ref

Reply:
added weak_ref

---
In reply to: [494522605](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r494522605) [](ancestors = 494522605)